### PR TITLE
Node Zealot

### PR DESCRIPTION
**Pull request by jameskerr**
A little bit of code and now we don't need deno. Zealot runs in the browser and in node. 

The test folder has been re-organized:

```
test
├── api/
│   ├── helpers/
│   ├── setup/
│   ├── tests/
│   └── jest.config.js
├── integration/
│   ├── helpers/
│   ├── setup/
│   ├── tests/
│   ├── README.md
│   └── jest.config.js
├── shared/
│   ├── data/
│   ├── factories/
│   ├── responses/
│   └── debug.ts
└── unit/
    ├── __mocks__/
    ├── helpers/
    ├── setup/
    └── jest.config.js
```

I've utilized a feature of jest called "projects". It allows for multiple test suites (with different configs) in a single repo.

https://jestjs.io/docs/configuration#projects-arraystring--projectconfig
https://homoly.me/posts/organizing-tests-with-jest-projects/

We can run tests like this now:

```
npx jest --projects test/unit
npx jest --projects test/api
npx jest --projects test/integration
```

To do:

- [x] Remove dependency on deno in ci
- [x] Fix the generate responses script

Not blocking todos:

- [ ] Add server handlers
- [ ] Remove mock zealot

### CODE REVIEW

File: app/core/utils/get-repo.test.ts [hunk 1]
```diff
@@ -1,7 +0,0 @@
-import {getRepo} from "./get-repo"
-
-test("returns whatever is in package.json", async () => {
-  expect(await getRepo("https://github.com/brimdata/brim")).toBe(
-    "brimdata/brim"
-  )
-})
```
Comment (jameskerr) [inline app/core/utils/get-repo.test.ts:7]: These files have been deleted and moved to `app/ipc/meta.ts`

File: zealot/ndjson/pipe_text.ts [hunk 2]
```diff
@@ -15,3 +28,49 @@ export async function* pipeText(stream: ReadableStream<Uint8Array> | null) {
...
+    next(): Promise<IteratorResult<string, void>> {
+      if (pending.length) {
+        return Promise.resolve(pending.shift())
+      } else {
+        return new Promise((res) => {
+          resolves.push(res)
+        })
+      }
+    },
+    [Symbol.asyncIterator]: function() {
```
Comment (mason-fish) [inline zealot/ndjson/pipe_text.ts:72]: 💯

File: zealot/ndjson/pipe_text.ts [hunk 3]
```diff
@@ -1,5 +1,18 @@
-export async function* pipeText(stream: ReadableStream<Uint8Array> | null) {
-  if (!stream) return
+export function pipeText(
+  stream: ReadableStream<Uint8Array> | null | NodeJS.ReadableStream
+): AsyncGenerator<string, void> {
+  if (!stream) return noop()
+  // @ts-ignore
+  if (stream.getReader) {
```
Comment (mason-fish) [inline zealot/ndjson/pipe_text.ts:6]: glad this ended up being easier to work around in the end!

File: test/shared/data/index.ts [hunk 4]
```diff
@@ -0,0 +1,17 @@
+import path from "path"
+import fs from "fs-extra"
+class Data {
+  getPath(name) {
+    return path.join(__dirname, name)
+  }
+
+  getDOMFile(name) {
```
Comment (mason-fish) [inline test/shared/data/index.ts:8]: Why is `DOM` in the name here? (i.e. why not just `getFile`?)

**Review by mason-fish — APPROVED** [review]
Lgtm, just one naming question. Not a blocker at all. Awesome work!!

File: test/shared/data/index.ts (hunk 4 above)
Comment (jameskerr) [reply test/shared/data/index.ts:8]: I was trying to distinguish between a [Web Api File](https://developer.mozilla.org/en-US/docs/Web/API/File) class (DOMFile) and a node file, or a brim.file? 

It's not often that we interact with a File instance from the Web APIs so it seemed like calling it out somehow would be useful.

I'm open to a different name though!

**Pull request merged**